bundles/sirix-core/src/main/java/org/sirix/service/json/serialize/JsonSerializerProperties.java

```diff
@@ -0,0 +1,191 @@
+import java.util.concurrent.ConcurrentMap;
+import org.sirix.utils.LogWrapper;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <h1>XMLSerializerProperties</h1>
+ *
+ * <p>
+ * XMLSerializer properties.
+ * </p>
+ *
+ * @author Johannes Lichtenberger, University of Konstanz
+ *
+ */
+public final class JsonSerializerProperties {
+
+  /** Logger. */
+  private static final LogWrapper LOGGER = new LogWrapper(LoggerFactory.getLogger(JsonSerializerProperties.class));
+
+  // ============== Class constants. =================
+
+  /** NO maps to false. */
+  private static final boolean NO = false;
+
+  // ============ Shredding constants. ===============
+
+  /** Serialization parameter: yes/no. */
+  public static final Object[] S_INDENT = {"indent", NO};
+
+  /** Specific serialization parameter: number of spaces to indent. */
+  public static final Object[] S_INDENT_SPACES = {"indent-spaces", 2};
+
+  /** Property file. */
+  private String mFilePath;
+
+  /** Properties. */
+  private final ConcurrentMap<String, Object> mProps = new ConcurrentHashMap<String, Object>();
+
+  /**
+   * Constructor.
+   */
+  public JsonSerializerProperties() {
+    try {
+      for (final Field f : getClass().getFields()) {
+        final Object obj = f.get(null);
+        if (!(obj instanceof Object[])) {
+          continue;
+        }
+        final Object[] arr = (Object[]) obj;
+        mProps.put(arr[0].toString(), arr[1]);
+      }
+    } catch (final IllegalArgumentException | IllegalAccessException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * <h2>Read properties</h2>
+   *
+   * <p>
+   * Read properties file into a concurrent map. Format of properties file:
+   * </p>
+   *
+   * <ul>
+   * <li>xmldecl=yes (possible values: yes/no)</li>
+   * <li>indent=no (possible values: yes/no)</li>
+   * <li>indent-spaces=2 (possible values: Integer)</li>
+   * <li>serialize-rest=no (possible values: yes/no)</li>
+   * <li>serialize-id=no (possible values: yes/no)</li>
+   * </ul>
+   *
+   * <p>
+   * Note that currently all properties have to be set. If specific key/value pairs are specified more
+   * than once the last values are preserved, so the default values are overridden by user specified
+   * values.
+   * </p>
+   *
+   * @param filePath path to properties file
+   * @return ConcurrentMap which holds property key/values.
+   */
+  public ConcurrentMap<String, Object> readProps(final String filePath) {
+    mFilePath = filePath;
+    if (!new File(mFilePath).exists()) {
+      throw new IllegalStateException("Properties file doesn't exist!");
+    }
+
+    try {
+      // Read and parse file.
+      final BufferedReader buffReader = new BufferedReader(new FileReader(mFilePath));
+      for (String line = buffReader.readLine(); line != null; line = buffReader.readLine()) {
+        line = line.trim();
+        if (line.isEmpty()) {
+          continue;
+        }
+
+        final int equals = line.indexOf('=');
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 125 does not find the substring and returns -1, the call to substring() on line 130 will throw IndexOutOfBoundsException.